### PR TITLE
Split docs on modality

### PR DESCRIPTION
**Pull request by stevhliu**
Currently, audio and computer vision lack content or the existing content is mixed with NLP. This PR splits the `toctree` on modality to make it easier to discover content for audio/computer vision while also allowing us to also scale to any additional modalities we want to support. As we create additional content, these new sections make it easier to collect specific content in one place. For example, the upcoming `generate` docs can be placed in the NLP section.

This structure can also help us identify gaps in the docs between each modality to ensure documentation is complete. For example, NLP has a page about tokenizers, and we can create a similar page for the other modalities using feature extractors and processors.

Other sections include:

- General usage for modality-neutral content.
- Performance and scalability for content related to large models.
- Contribute for how to test, open a PR, and add models/pipelines.

After we split the docs, the next step would be to start planning and creating additional content to make the audio/computer vision sections more complete.

Looking forward to hearing what you think :)

### CODE REVIEW

**Comment by sgugger** [comment]
Not really convinced by this as it's now very unclear that all the task tutorials are task-specific tutorials. I liked it better when they were grouped altogether under task. It also backfires from the intent as this shows we don't really support vision (one entry) or speech (two entries) compared to NLP (9 entries).

This revamp also puts advanced guides on the top when they should be way lower in the table of contents (such as Benchmarks or Migrating from other packages).

**Comment by stevhliu** [comment]
Thanks for the feedback! 🤗

> Not really convinced by this as it's now very unclear that all the task tutorials are task-specific tutorials.

Hmm, do you mean it’s more unclear now because each task tutorial is separated by modality? This seems clearer to me since the section headers are more scannable.

> It also backfires from the intent as this shows we don't really support vision (one entry) or speech (two entries) compared to NLP (9 entries).

Good perspective, and I totally see what you mean! I think another way to look at it is by creating these new sections, we’re signaling that we plan to create more content for audio/CV. This gives these sections more prominence, which shows we want to focus on audio/CV. So even though it looks pretty bare right now, I think that’s ok since these sections will grow.

> This revamp also puts advanced guides on the top when they should be way lower in the table of contents (such as Benchmarks or Migrating from other packages).

I don’t think we should put guides lower because they are more advanced. Instead, it may be better to prioritize guides that users are more likely to find useful. For example, I think the Migration/Train with a script guides are pretty useful. This may be a symptom of how I grouped all these guides under General Usage, in which case, we can try breaking up the section and reordering these guides by their utility.

**Comment by stevhliu** [comment]
Ok I see now! You're worried users won't know the task-specific guides are guides about fine-tuning a model for a task if it is just thrown into the NLP section. I think there are some things we can do to help make this clearer to users (in order of preference):

1. Include an overview page for each modality section explaining what users can expect to find.
2. Update the task-specific guides to have clearer titles like, How to fine-tune for text classification.
3. Create another nested section in each modality that focuses on the task-specific guides.

> Benchmarks or migrating are both advanced and not useful (benchmarks have 0 issues and we are even questioning whether they should stay in the library and pytorch-pretrained-bert ceased to exist a while ago).

For sure! Benchmark, Migration, and Troubleshoot are bottom-3 in page views in the General Usage section. I can bump these out and move them closer to the bottom.

**Comment by sgugger** [comment]
Option 2 or 3 are good compromise (my preference goes to 3 if nested-ness is not an issue). I'd leave Troubleshoot in the General Usage section (hopefully we can make it better so it gets more views), but yeah, the other two are out of place there IMO.

Let's see what other people think as well, @LysandreJik @patrickvonplaten to name a few :-)

**Comment by LysandreJik** [comment]
I also think that option 3) sounds like the best approach. I don't have a problem with adding a nesting level.

**Comment by stevhliu** [comment]
I nested the NLP section but it looks a little off since the content inside isn't aligned on the same level (I pinged @mishig25 on this). I didn't add a nested level for the audio and image sections since there's no content in those sections yet, and it might look a little strange.

**Comment by stevhliu** [comment]
Hi team, just wanted to circle back on this and see if there are any more comments or feedback about how the docs are split.  Otherwise, I think we're ready to merge! 🙂

**Comment by patrickvonplaten** [comment]
Option 3.) Also looks like the right one to me :-) 

However I'm not a big fan of "Image" as a title. Could we maybe try to align those sections a bit with how we call the modalities on the Hub: https://huggingface.co/tasks -> so maybe replace "Image" with "Computer Vision"?

Wdyt @sgugger @LysandreJik @osanseviero

File: docs/source/en/_toctree.yml [hunk 1]
```diff
@@ -21,44 +21,55 @@
...
+        title: Language modeling
+      - local: tasks/translation
+        title: Translation
+      - local: tasks/summarization
+        title: Summarization
+      - local: tasks/multiple_choice
+        title: Multiple choice
+      title: Task guides
+      isExpanded: false
+    title: "NLP"
```
Comment (patrickvonplaten) [inline docs/source/en/_toctree.yml:62]: (nit) Think it's actually nicer to write it out "Natural Language Processing" , but both ways are fine with me!

File: docs/source/en/_toctree.yml [hunk 2]
```diff
@@ -21,44 +21,55 @@
...
-    title: Train with a script
-  - local: sagemaker
-    title: Run training on Amazon SageMaker
-  - local: multilingual
-    title: Inference for multilingual models
-  - local: converting_tensorflow_models
-    title: Converting TensorFlow Checkpoints
-  - local: serialization
-    title: Export 🤗 Transformers models
+    title: "Image"
```
Comment (patrickvonplaten) [inline docs/source/en/_toctree.yml:72]: Think we should call it "Computer Vision" to align with https://huggingface.co/tasks

File: docs/source/en/_toctree.yml [hunk 3]
```diff
@@ -21,44 +21,55 @@
...
+      title: Train with a script
+    - local: sagemaker
+      title: Run training on Amazon SageMaker
+    - local: converting_tensorflow_models
+      title: Converting TensorFlow Checkpoints
+    - local: serialization
+      title: Export 🤗 Transformers models
+    - local: troubleshooting
+      title: Troubleshoot
+    title: "General usage"
```
Comment (sgugger) [inline docs/source/en/_toctree.yml:39]:
```suggestion
    title: General usage
```
The quotes create a diff when we load and save of the toctree with yaml.

File: docs/source/en/_toctree.yml [hunk 4]
```diff
@@ -21,44 +21,55 @@
...
+        title: Language modeling
+      - local: tasks/translation
+        title: Translation
+      - local: tasks/summarization
+        title: Summarization
+      - local: tasks/multiple_choice
+        title: Multiple choice
+      title: Task guides
+      isExpanded: false
+    title: "Natural Language Processing"
```
Comment (sgugger) [inline docs/source/en/_toctree.yml:62]:
```suggestion
    title: Natural Language Processing
```

File: docs/source/en/_toctree.yml [hunk 5]
```diff
@@ -21,44 +21,55 @@
...
+  - sections:
     - local: tasks/audio_classification
       title: Audio classification
     - local: tasks/asr
       title: Automatic speech recognition
+    title: "Audio"
```
Comment (sgugger) [inline docs/source/en/_toctree.yml:68]:
```suggestion
    title: Audio
```

File: docs/source/en/_toctree.yml [hunk 6]
```diff
@@ -21,44 +21,55 @@
...
-    title: Train with a script
-  - local: sagemaker
-    title: Run training on Amazon SageMaker
-  - local: multilingual
-    title: Inference for multilingual models
-  - local: converting_tensorflow_models
-    title: Converting TensorFlow Checkpoints
-  - local: serialization
-    title: Export 🤗 Transformers models
+    title: "Computer Vision"
```
Comment (sgugger) [inline docs/source/en/_toctree.yml:72]:
```suggestion
    title: Computer Vision
```

File: docs/source/en/_toctree.yml [hunk 7]
```diff
@@ -84,32 +95,32 @@
...
     - local: perf_hardware
       title: Custom hardware for training
-    title: Performance and scalability
-  - local: big_models
-    title: Instantiating a big model
+    - local: big_models
+      title: Instantiating a big model
+    - local: debugging
+      title: Debugging
+    title: "Performance and scalability"
```
Comment (sgugger) [inline docs/source/en/_toctree.yml:102]:
```suggestion
    title: Performance and scalability
```

File: docs/source/en/_toctree.yml [hunk 8]
```diff
@@ -84,32 +95,32 @@
...
+      title: How to contribute to transformers?
+    - local: add_new_model
+      title: How to add a model to 🤗 Transformers?
+    - local: add_new_pipeline
+      title: How to add a pipeline to 🤗 Transformers?
+    - local: testing
+      title: Testing
+    - local: pr_checks
+      title: Checks on a Pull Request
+    title: "Contribute"
```
Comment (sgugger) [inline docs/source/en/_toctree.yml:114]:
```suggestion
    title: Contribute
```

File: docs/source/en/_toctree.yml [hunk 9]
```diff
@@ -84,32 +95,32 @@
...
-  - local: add_new_model
-    title: How to add a model to 🤗 Transformers?
-  - local: add_new_pipeline
-    title: How to create a custom pipeline?
-  - local: testing
-    title: Testing
-  - local: pr_checks
-    title: Checks on a Pull Request
-  title: How-to guides
+  title: "How-to guides"
```
Comment (sgugger) [inline docs/source/en/_toctree.yml:123]:
```suggestion
  title: How-to guides
```

**Review by sgugger — APPROVED** [review]
LGTM barring the quotes to remove everywhere. Thanks for iterating on this PR!

**Pull request merged**